# Project: Super Store Sales

## Table of Contents
<ul>
<li><a href="#intro">Introduction</a></li>
<li><a href="#Wrangling">Data Wrangling</a></li>
<li><a href="#Exploratory">Exploratory Data Analysis</a></li>
<li><a href="#conclusions">Explanatory And Conclusions</a></li>
</ul>

<a id='intro'></a>
## Introduction

> I will analyze and explore Superstore Dataset which contains about 17000 observations and 26 variables From Jan 2010 to Dec 2013 for 3 departments: furniture, technolgy, and office suppliers throughout 50 Countries and 1523 cities around the world in total orders quantities about 440,000 orders with total profit about 15,000,000 million dollars. I used this dataset for practice. In this project, I will use Python in the wrangling and exploratory Processes and I will use Tableau for the explanatory process and visualizations. I will gather the dataset from the CSV file to the jupyter Notebook then I will do the wrangling process to handle the limitations and started to explore the data programmatically with Python. After cleaning the data to be ready for the explanatory process, I will export the cleaned data to CSV File then I will load it to Tableau.

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sb
%matplotlib inline
from scipy import stats
import datetime

<a id='wrangling'></a>
## Data Wrangling

In [2]:
Superstore = pd.read_csv('SuperstoreSalesTraining.csv')
print(Superstore.shape)
Superstore.sample(5)

(16798, 26)


Row Order Priority  Order Date  Order Discount  Unit Price  \
13446  13447         Medium  20/05/2013  87332       2%         6.0   
4801    4802            Low  18/05/2011  31907       0%         8.0   
10371  10372            Low  10/10/2012  89074       2%        15.0   
15284  15285           High  16/09/2013  57734      10%        17.0   
2715    2716       Critical  10/10/2010  28002      10%         9.0   

       Order Quantity   Sales  Profit  Shipping Cost  ... Customer_ID  \
13446              11   64.68   14.44            8.0  ...        3109   
4801               55  440.00  264.40            4.0  ...        2921   
10371               7  102.90   47.80           11.0  ...        3019   
15284              57  872.10  230.25           12.0  ...         849   
2715                9   72.90   22.30            2.0  ...        1601   

            Customer Name         Region              State  \
13446      Samuel Simmons  North America              Idaho   
4801   Dean Shapiro Britt        AsiaPac               Soul   
10371          Edwin Lang  North America         New Jersey   
15284        Vernon Stark           EMEA      �le-de-France   
2715          Glenn Gibbs          Latam  Provincia de Lima   

               Country / Region         City  Postal Code   Ship Date  \
13446  United States of America    Pocatello      83201.0  23/05/2013   
4801          Republic of Korea        Seoul          NaN  23/05/2011   
10371  United States of America  East Orange       7017.0  12/10/2012   
15284                    France        Paris      75007.0  18/09/2013   
2715                       Peru         Lima          NaN  12/10/2010   

         Ship Mode                                          SubRegion  
13446  Regular Air  West                                          ...  
4801   Regular Air                                                NaN  
10371  Regular Air  East                                          ...  
15284  Regular Air                                                NaN  
2715   Regular Air                                                NaN  

[5 rows x 26 columns]

In [3]:
Superstore.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16798 entries, 0 to 16797
Data columns (total 26 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Row                  16798 non-null  int64  
 1   Order Priority       16798 non-null  object 
 2   Order Date           16798 non-null  object 
 3   Order                16798 non-null  int64  
 4   Discount             16798 non-null  object 
 5   Unit Price           16798 non-null  float64
 6   Order Quantity       16798 non-null  int64  
 7   Sales                16798 non-null  float64
 8   Profit               16798 non-null  float64
 9   Shipping Cost        16798 non-null  float64
 10  Product Base Margin  16798 non-null  object 
 11  Department           16798 non-null  object 
 12  Container            16798 non-null  object 
 13  Category             16798 non-null  object 
 14  Item                 16798 non-null  object 
 15  Customer Segment     16798 non-null 

In [4]:
# Checking the duplicated observations.

Superstore.duplicated().sum()

0

In [5]:
# Dropping the row-column because it is just an index for the row number.

Superstore= Superstore.drop('Row', axis=1)

In [6]:
Superstore['Order'] = Superstore['Order'].astype('str')

In [7]:
# Showing the unseen columns.

hidden_Superstore = Superstore[['Department','Container','Category','Item','Customer Segment','Customer_ID']]
hidden_Superstore.sample(5)

Department   Container                        Category  \
15656  Office Supplies   Small Box                       Envelopes   
14870  Office Supplies   Small Box                          Labels   
2226         Furniture  Jumbo Drum              Chairs & Chairmats   
8909   Office Supplies   Small Box                          Labels   
13574  Office Supplies   Small Box  Binders and Binder Accessories   

                                                    Item Customer Segment  \
15656                     Brown Kraft Recycled Envelopes        Corporate   
14870                                          Avery 491   Small Business   
2226   Office Star Flex Back Scooter Chair with Alumi...   Small Business   
8909                                            Avery 52        Corporate   
13574                                3M Organizer Strips        Corporate   

       Customer_ID  
15656         2925  
14870         3188  
2226          2222  
8909          3090  
13574          368

In [8]:
Superstore.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16798 entries, 0 to 16797
Data columns (total 25 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Order Priority       16798 non-null  object 
 1   Order Date           16798 non-null  object 
 2   Order                16798 non-null  object 
 3   Discount             16798 non-null  object 
 4   Unit Price           16798 non-null  float64
 5   Order Quantity       16798 non-null  int64  
 6   Sales                16798 non-null  float64
 7   Profit               16798 non-null  float64
 8   Shipping Cost        16798 non-null  float64
 9   Product Base Margin  16798 non-null  object 
 10  Department           16798 non-null  object 
 11  Container            16798 non-null  object 
 12  Category             16798 non-null  object 
 13  Item                 16798 non-null  object 
 14  Customer Segment     16798 non-null  object 
 15  Customer_ID          16798 non-null 

In [9]:
# Converting the date columns to datetype.

Superstore['Order Date'] = pd.to_datetime(Superstore['Order Date'], dayfirst=True).astype('datetime64[ns]')
Superstore['Ship Date'] = pd.to_datetime(Superstore['Ship Date'], dayfirst=True).astype('datetime64[ns]')

In [10]:
# Converting the other columns to the correct datatype.

Superstore['Postal Code'] = Superstore['Postal Code'].astype('str')
Superstore['Order Priority'] = Superstore['Order Priority'].astype('category')
Superstore['Order'] = Superstore['Order'].astype('str')
Superstore['Department'] = Superstore['Department'].astype('category')
Superstore['Container'] = Superstore['Container'].astype('category')
Superstore['Category'] = Superstore['Category'].astype('category')
Superstore['Customer Segment'] = Superstore['Customer Segment'].astype('category')
Superstore['Customer_ID'] = Superstore['Customer_ID'].astype('str')
Superstore['Region'] = Superstore['Region'].astype('category')
Superstore['Ship Mode'] = Superstore['Ship Mode'].astype('category')
Superstore['SubRegion'] = Superstore['SubRegion'].astype('category')

In [11]:
Superstore.sample(5)

Order Priority Order Date  Order Discount  Unit Price  Order Quantity  \
2473             Low 2010-09-16  90385       5%       442.0              14   
12139            Low 2013-02-27  86198       7%       126.0               4   
631    Not Specified 2010-03-05  20773       8%        11.0              44   
5276        Critical 2011-07-03  87622       7%        37.0               7   
8464          Medium 2012-04-19  90980      10%        90.0               2   

         Sales   Profit  Shipping Cost Product Base Margin  ... Customer_ID  \
2473   5878.60  3140.88           15.0              56.00%  ...        2157   
12139   468.72   253.08            9.0              59.00%  ...         555   
631     445.28   142.20            3.0              38.00%  ...        1013   
5276    240.87    74.06           14.0              41.00%  ...        2931   
8464    162.00    58.80           42.0              66.00%  ...        1103   

             Customer Name         Region                      State  \
2473   Tom Hoyle Honeycutt  North America                   Michigan   
12139         Walter Young  North America                       Utah   
631              Joel Mann        AsiaPac  R�gion Capitale Nationale   
5276            Faye Hanna  North America                 California   
8464        Sidney Bowling  North America                   Nebraska   

               Country / Region             City Postal Code  Ship Date  \
2473   United States of America           Warren     48093.0 2010-09-25   
12139  United States of America   Pleasant Grove     84062.0 2013-03-01   
631                 Philippines           Manila         nan 2010-03-06   
5276   United States of America  El Dorado Hills     95630.0 2011-07-06   
8464   United States of America            Omaha     68046.0 2012-04-20   

            Ship Mode                                          SubRegion  
2473   Delivery Truck  Central                                       ...  
12139     Express Air  West                                          ...  
631       Regular Air                                                NaN  
5276      Regular Air  West                                          ...  
8464   Delivery Truck  Central                                       ...  

[5 rows x 25 columns]

<a id='Exploratory'></a>
## Exploratory Data Analysis

In [12]:
Superstore['Order Priority'].value_counts(ascending = False)

High             3536
Low              3440
Not Specified    3344
Medium           3262
Critical         3216
Name: Order Priority, dtype: int64

In [13]:
Superstore['Department'].value_counts(ascending = False)

Office Supplies    9220
Technology         4130
Furniture          3448
Name: Department, dtype: int64

In [14]:
Superstore['Container'].value_counts(ascending = False)

Small Box     8694
Wrap Bag      2336
Small Pack    1912
Jumbo Drum    1248
Jumbo Box     1064
Large Box      812
Medium Box     732
Name: Container, dtype: int64

In [15]:
Superstore['Category'].value_counts(ascending = False)

Paper                             2450
Binders and Binder Accessories    1830
Telephones and Communication      1766
Office Furnishings                1576
Computer Peripherals              1516
Pens & Art Supplies               1266
Storage & Organization            1092
Appliances                         868
Chairs & Chairmats                 772
Tables                             722
Office Machines                    674
Labels                             576
Envelopes                          492
Bookcases                          378
Rubber Bands                       358
Scissors, Rulers and Trimmers      288
Copiers and Fax                    174
Name: Category, dtype: int64

In [16]:
Superstore['Customer Segment'].value_counts(ascending = False)

Corporate         6152
Home Office       4064
Consumer          3298
Small Business    3284
Name: Customer Segment, dtype: int64

In [17]:
Superstore['Region'].value_counts(ascending = False)

North America    9482
AsiaPac          3802
EMEA             1894
Latam            1620
Name: Region, dtype: int64

In [18]:
Superstore['State'].value_counts(ascending = False).head(10)

California                  1021
Texas                        646
Illinois                     584
New York                     574
Florida                      522
Guangdong Sheng              417
Ohio                         396
Province de Buenos Aires     341
Michigan                     327
Washington                   327
Name: State, dtype: int64

In [19]:
Superstore['Country / Region'].value_counts(ascending = False).head(10)

United States of America    9426
China                       1257
India                        746
Brazil                       672
Japan                        507
Mexico                       388
Argentina                    341
Republic of Korea            325
Egypt                        265
France                       242
Name: Country / Region, dtype: int64

In [20]:
Superstore['Ship Mode'].value_counts(ascending = False)

Regular Air       12540
Delivery Truck     2292
Express Air        1966
Name: Ship Mode, dtype: int64

In [21]:
Superstore['SubRegion'].value_counts(ascending = False)

Central                                                                                                                                                                                                                                                            2899
East                                                                                                                                                                                                                                                               2289
West                                                                                                                                                                                                                                                               2284
South                                                                                                                                                                                                           

In [22]:
# Converting the Discount & Product Base Margin to floats instead of objects.

Superstore['Discount'] = Superstore['Discount'].str.rstrip('%').astype('float') / 100.0
Superstore['Product Base Margin'] = Superstore['Product Base Margin'].str.rstrip('%').astype('float') / 100.0

In [23]:
print(Superstore.shape)
Superstore.sample(5)

(16798, 25)


Order Priority Order Date  Order  Discount  Unit Price  Order Quantity  \
901           Medium 2010-04-01  17795      0.06        11.0               1   
9603          Medium 2012-07-30  89949      0.08        12.0              14   
15853       Critical 2013-10-24   4835      0.04         3.0               6   
10252           High 2012-10-02   3460      0.08         6.0              53   
8309            High 2012-04-08  52929      0.04        16.0              48   

        Sales  Profit  Shipping Cost  Product Base Margin  ... Customer_ID  \
901     10.34   -2.04            6.0                 0.42  ...        3305   
9603   154.56  -13.72            7.0                 0.04  ...        1775   
15853   17.28   12.04            2.0                 0.82  ...        1256   
10252  292.56  185.80            5.0                 0.68  ...        3079   
8309   737.28  336.60            9.0                 0.49  ...        2408   

       Customer Name         Region         State          Country / Region  \
901     Susan Bowers          Latam    Sa?o Paulo                    Brazil   
9603    Marlene Kirk  North America       Indiana  United States of America   
15853    Neil Curtis        AsiaPac         Tokyo                     Japan   
10252  Andrew Levine  North America  Pennsylvania  United States of America   
8309    Billie Parks  North America       Ontario                    Canada   

               City Postal Code  Ship Date    Ship Mode  \
901       Sao Paulo         nan 2010-04-03  Regular Air   
9603     South Bend     46614.0 2012-07-31  Regular Air   
15853         Tokyo         nan 2013-10-24  Regular Air   
10252  Philadelphia     19112.0 2012-10-04  Regular Air   
8309        Toronto         nan 2012-04-09  Express Air   

                                               SubRegion  
901                                                  NaN  
9603   Central                                       ...  
15853                                                NaN  
10252  East                                          ...  
8309   Canada                                        ...  

[5 rows x 25 columns]

In [24]:
hidden_Superstore = Superstore[['Department','Container','Category','Item','Customer Segment','Customer_ID']]
hidden_Superstore.sample(5)

Department   Container              Category  \
334    Office Supplies   Small Box                 Paper   
9138   Office Supplies   Small Box             Envelopes   
15892       Technology   Small Box  Computer Peripherals   
4626         Furniture  Small Pack    Office Furnishings   
12235       Technology  Small Pack  Computer Peripherals   

                                          Item Customer Segment Customer_ID  
334                                  Xerox 194        Corporate          24  
9138   #10- 4 1/8" x 9 1/2" Recycled Envelopes      Home Office        1460  
15892           Micro Innovations 104 Keyboard   Small Business        1674  
4626                   Nu-Dell Executive Frame        Corporate        2000  
12235        Hewlett-Packard 4.7GB DVD+R Discs         Consumer        1753

In [25]:
Superstore.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16798 entries, 0 to 16797
Data columns (total 25 columns):
 #   Column               Non-Null Count  Dtype         
---  ------               --------------  -----         
 0   Order Priority       16798 non-null  category      
 1   Order Date           16798 non-null  datetime64[ns]
 2   Order                16798 non-null  object        
 3   Discount             16798 non-null  float64       
 4   Unit Price           16798 non-null  float64       
 5   Order Quantity       16798 non-null  int64         
 6   Sales                16798 non-null  float64       
 7   Profit               16798 non-null  float64       
 8   Shipping Cost        16798 non-null  float64       
 9   Product Base Margin  16798 non-null  float64       
 10  Department           16798 non-null  category      
 11  Container            16798 non-null  category      
 12  Category             16798 non-null  category      
 13  Item                 16798 non-

In [ ]:
# Storing the cleaed dataset

Superstore.to_csv('Super Store.csv', index=False)

<a id='conclusions'></a>
## Explanatory And Conclusions

### Almost, The datasets were ready and I did small cleaning operations. After cleaning the data I started to pose questions and analyze data Throght Tableau to find answers via this link: 

https://public.tableau.com/app/profile/khaled.yaseen/viz/SuperStore_16449743620740/GenralDashboard

##### Q1: What is the most Country in total Profit?
- USA is the highest country made the largest total profit for the super store with about 4.1 million dollars then China with about 1.8 million dollars.

##### Q2: USA is the largest Country in Total orders in different Cities, Is USA Is the largest Country in Avg profit Also?
- No, USA is not the largest, USA has Avg profit 437 dollars per order for 9426 Orders, The largest in Avg is Moroco with 2383 dollars per order for only 22 orders then 2284 dollars per order for only 45 orders then norway with 2162 dollars per order for only 11 orders. From the map the middle east and the eauropean countries  have high avg profit values, Also China has 1429 Avg profit per order for 1257 orders and that is a high order Count so I think the superstore should make a true interest for other countries like the interest for USA! . But there is an important queation for the shipping cost for these countries!

##### Q3: What is the Avg shipping cost for countries and is there a gap difference in the cost from USA?
- The Avg shipping cost for USA 12.81 dollars per order, For Moroco 12.86 dollars!, For iraq 15.8, For Norway 24.09, For china 13.28, For ukraine 11.25 dollars which has Avg profit 2027 dollars per orders for only 24 orders, For algeria only 8.17 dollars which has Avg profit 1379 dollars per order for only 18 orders. so, I think the superstoere should to target other markets beside the united states!.

##### Q4: What are the cities with the highest profit?
- The first one is Buenos Aires from Argentia with total profit 560,000 dollars then Guangzhou From China with 535,000 dollars, Cairo comes in the 5th place with 366,000 dollars!. the first city comes from USA is Los Anglos in the 11th place with total profit 286,000 dollars. And that is another evidence that the superstore should expand to another marketplaces.

##### Q5: Which item has the largest in the profit and in which department?
- The item of global troy of excutive leather low back tilter in the furniture department has the most totaal total profit of 277,000 dollars and its orders count are 34 orders with Avg product base margin is 0.6. The department of furniture has the biggest share in the highest items in profit!

##### Q6: What is the largest category and department in profit?
- The telephones and communications in the technology department has the largest profit with 2,436,000 dollars and its order counts are 1766 orders with avg product base margin 0.58.

##### Q7: What are the order count perctiles for departments?
- The furniture has 39.61%, the technolgy has 38.68%, and the office suppliers has 21.71%.

##### Q8: What is the total profit for each departmesnt?
- The furniture has 5,870,000 dollars, the technolgy has 5,731,000 dollars, and the office suppliers has 3,217,000.

##### Q9: What is the Avg product margin for each department?
- The furniture has 0.5837, the technolgy has 0.5515, and the office suppliers has 0.4468.

##### 10: What is the Avg Unit Price for each department?
- The technolgy is the largest with 181.4 dollars then the furniture with 121.5 dollars then the office suppliers with 36.1 doolars per order.

##### 11: What is the category is the largest in the Avg Unit Price?
- It is copiers and fax from the technolgy department with 746 dollars then technolgy&oofice machines also from the technolgy department with 565.7 dollars per order.

##### 12: When did the largest profit happen and in each category? In that perid, Is this category has the largest number in order count?
- The largest profit happened in Nov 2012 with total profit 164,000 dollars and only 20 orders in tables category but in this month paper has the largest number in order counts with 54 orders and 6874 dollars!.The largest order count happened in Oct 2013 with 95 orders and only total profit 12,882 dollars! but in this month telephones and communication has the largest profits 82,299 and only 38 orders!. In general paper has the largest order counts over years, the largest profit varies between furniture and technology categories over years, and almost the largest profit happens in Nov from every year!. Wow, This led me to check is there are discounts in december?

##### 13: How the discounts were over year and there were special discounts in november?
- Another surprise for me, Tables has largest discounts over years with a gab from other categories!. But i didn't found any special discounts in november!. (See Q15)

##### 14: For Departments, What was the time anlaysis?
- Like the output from Categories, Office suppliers has the largest number in order counts, the most happened in Oct 2013 with 299 orders. The furniture and technology have the largest number in profit and the largest profit happened for furniture in Nov 2012 With total 365,000 dollars.

##### 15: What are the average discount profit and order counts for different discounts?
- Another insight supporst the insight from Q13, Only tables and furnishings have discounts over than 10% . All other categories thier discounts less than 10%.

##### 16: What are the largest category in Avg Shipping cost per orders?
- As expexted, Furniture categories are the largest With Avg 30.89 dollars for order, Tables is the highest in Avg Shipping cost with 57,36 dollars. Technology and office supplies almost the same in avg Shipping cost

##### 17: What are customer segment percentiles?
- Corporate with 36.8% of order counts, Home Office with 24.1%, Small business with 19.74%, then Consumer with 19.36%.

##### Q18: What is the total profit from each custome segment?
- Superstore got from Corporate with 5,453,000 dollars, From Home Office customers 3,572,000 dollars, From Small business 2,925 dollars, From Consumer 2,869,000 dollars. 
##### 19: When did the largest profit happen from customers and from which customer? In that perid, Is this customer has the largest number in order count?
- The Largest profit came from Corporate in Nov 2012 with total profit 274,000 dollars more little than small business in the same month which has 267,000 dollars, And Yes Corporate has the largest order counts in the same month with 197 orders. In general corporate has the largest profit and order counts over years.

##### 20: Who is the customer with the largest profit ?
- It is Joan Oakley Shaafer corporate with total profit 107,000 dollars and it has 26 orders. 

##### 21: What are the average delivery periods for different order priorities ?
- Low Priority takes more than 4 days to deliver the orders. Other priorities took form 1 to 2 days.

##### 22: What are the percentiles for different order priorities ?
- Almost the same range from 19 to 21% for the 4 priorities.

##### 23: What are the profit percentiles for different order priorities ?
- It ranges from 17 to 23.5% for the 4 priorities.

##### 24: What are the containers percentiles ?
- Small Box is the largest with more than 50% in compared to the other containers.

##### 25: What are the value and the average unite price for goods inside containers ?
- large box contain the most valuable goods with average unite price 347.5 dollars.

##### 26: What are the ship mode percentiles ?
- Regular air is the largest with almost 75%.

##### 27: What are the  avg shipping cost for ship modes ?
- Delivery Truck is the largest with 45.39 Dollars for Avg cost.

##### 28: What do the  ship mode and department  occur most?
- It is small box in regular air happened almost 750 times.

##### 29: What are the correlations between order quantity, shipping cost and profit?
- They are postive correlations.

##### 30: What Is the most popular Unit Price range?
- It is from 0 to 1000 dollars.

##### 30: What Is the most popular Product base margin range?
- It is from 0.35 to 0.65 and the most profit occurres on products base margin from 0.55 to 0.6.

##### 31: What Is the relation between product margins and Avg discount?
- I finish the report with the last insightful finding, the product with high margins above 0.6 release high discounts to encourage buying especially in the furniture department. 